d2compiler/compile.go

```diff
@@ -504,6 +528,29 @@ func (c *compiler) compileEdge(obj *d2graph.Object, e *d2ir.Edge) {
 	}
 }
 
+func (c *compiler) compileEdgeMap(edge *d2graph.Edge, m *d2ir.Map) {
+	class := m.GetField("class")
+	if class != nil {
+		className := class.Primary()
+		if className == nil {
+			c.errorf(class.LastRef().AST(), "class missing value")
+		} else {
+			classMap := m.GetClosestClassMap(className.String())
+			if classMap != nil {
+				c.compileEdgeMap(edge, classMap)
```

Comment:
What if a class contains `class: X` where X is the ID of the class itself?

Reply:
i don't understand. can you say again?

Reply:
```
classes: {
  first: {
    class: first
  }
}

my shape: {
  class: first
}
```

Will that cause an infinite loop?

Reply:
ohh hm i think we should prevent `class` from appearing in `classes`. 

i thought maybe they'd want to use it to construct new classes off a template, but that's not how classes are used. you'd just apply two classes to one shape to achieve that

Reply:
Sounds good to me.